### ISSUE TITLE
When deselecting multiple rows the last one is not getting unselected

### ISSUE THREAD

**Issue by demetris-manikas** [DataTables/Select#47]
Hi.
Playing around with the plugin I noticed that when deselecting multiple items the last one remains selected.
I dug a bit into the code and noticed the following code that intentionally prevents the last row from being deselected.  
https://github.com/DataTables/Select/blob/36e53e42f5220e98879d2554551aabbfaba7d0ff/js/dataTables.select.js#L695-L697

I checked the behavior of many programs and none acts like that.
Can you please explain the reasoning behind it?

Anyway I believe that this behavior should be configurable what is yours?
Thanks.

**Comment by AllanJard** [comment]
Just to confirm what you mean - if we take [this example](https://datatables.net/extensions/select/examples/initialisation/simple.html):

1. Click on the row "Angelica Ramos" to select it.
2. Shift click on the row "Bruno Nash" to select it and the rows between the two (6 rows now selected)
3. Shift click on the row "Bradley Greer" - that row will remain selected (3 rows now selected in the table)
![image](https://user-images.githubusercontent.com/84905443/146897778-cbb2e5b7-03ac-4c8e-a04b-7dfbeac233ce.png)

Are you expecting the Bradley row to also deselect? I've just tried that in Gnome's file browser and it operates the same way. I don't have a Mac or Windows machine here at the moment, but I think they operate the same way?

**Comment by demetris-manikas** [comment]
Yes that is exactly what I mean. I tried OpenOffice calc and gmail and they don't operate that way, on the other hand the file manager on my Xubuntu does.
Obviously there are different approaches to the behavior and that is why I propose that there has to be a way to switch between them.
I use the multi+shift variant for selection and it really feels weird when multi-select and multi-deselect are not affecting the same rows.

**Comment by AllanJard** [comment]
Its an interesting one this. I kind of want to have just a single behaviour to try and get some consistency. I see what you mean about Gmail. I just tried it there too and I think that does feel a bit more natural.

I'm fairly sure that when I wrote this originally I tried it on Win / Mac / Gnome / Kde and they all operated the same way for their file managers, which is why I went with this method. We'll have a chat about it here in the office and feedback when done (which will likely be after the holidays).
